source/rdt/rdt_amd64.go

```diff
@@ -0,0 +1,98 @@
+// +build amd64
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rdt
+
+type CpuidRet struct {
+	EAX, EBX, ECX, EDX uint32
+}
+
+const (
+	// CPUID EAX input values
+	LEAF_EXT_FEATURE_FLAGS = 0x07
+	LEAF_RDT_MONITORING    = 0x0f
+	LEAF_RDT_ALLOCATION    = 0x10
+
+	// CPUID ECX input values
+	RDT_MONITORING_SUBLEAF_L3 = 1
+
+	// CPUID bitmasks
+	EXT_FEATURE_FLAGS_EBX_RDT_M                                 = 1 << 12
+	EXT_FEATURE_FLAGS_EBX_RDT_A                                 = 1 << 15
+	RDT_MONITORING_EDX_L3_MONITORING                            = 1 << 1
+	RDT_MONITORING_SUBLEAF_L3_EDX_L3_OCCUPANCY_MONITORING       = 1 << 0
+	RDT_MONITORING_SUBLEAF_L3_EDX_L3_TOTAL_BANDWIDTH_MONITORING = 1 << 1
+	RDT_MONITORING_SUBLEAF_L3_EDX_L3_LOCAL_BANDWIDTH_MONITORING = 1 << 2
+	RDT_ALLOCATION_EBX_L3_CACHE_ALLOCATION                      = 1 << 1
+	RDT_ALLOCATION_EBX_L2_CACHE_ALLOCATION                      = 1 << 2
+	RDT_ALLOCATION_EBX_MEMORY_BANDWIDTH_ALLOCATION              = 1 << 3
+)
+
+func discoverRDT() []string {
+	features := []string{}
+
+	// Read cpuid information
+	extFeatures := cpuid(LEAF_EXT_FEATURE_FLAGS, 0)
+	rdtMonitoring := cpuid(LEAF_RDT_MONITORING, 0)
+	rdtL3Monitoring := cpuid(LEAF_RDT_MONITORING, RDT_MONITORING_SUBLEAF_L3)
+	rdtAllocation := cpuid(LEAF_RDT_ALLOCATION, 0)
+
```

Comment:
Looks better now. One last concern that I have is separation of Intel/AMD CPUs within amd64 build target. Maybe it should be one more "safety net" to check for vendor id before trying to execute cpuid assembly instruction?

Reply:
The patch replicates the current logic. This should work fine on AMD, too (i.e. not report any RDT features) as those are Intel-specific leafs, reserved/not used in AMD cpus